src/component/providers/SWRProvider/SWRProvider.tsx

```diff
@@ -22,17 +22,23 @@ const SWRProvider: React.FC<ISWRProviderProps> = ({
     const handleFetchError = error => {
         setShowLoader(false);
         if (error.status === 401) {
-            cache.clear();
             const path = location.pathname;
             // Only populate user with authDetails if 401 and
             // error is not invalid token
             if (error?.info?.name !== INVALID_TOKEN_ERROR) {
                 mutate(USER_CACHE_KEY, { ...error.info }, false);
             }
-            if (path === '/login') {
+
+            if (
+                path === '/login' ||
+                path === '/new-user' ||
+                path === '/reset-password'
+            ) {
                 return;
             }
 
+            cache.clear();
+
             history.push('/login');
```

Comment:
I just got here and this code is new to me butt is this still correct even though the check has changed from just /login to checking against any of /login, /new-user and /reset-password?

Reply:
Yeah. So if you are on any of those pages, then we want to keep whatever is in the useSWR cache because the cache represents every getter hooks cache, and we use the cache to keep track of data fetched from the API like `useResetPassword`. If we are not on any of those pages, then we clear the entire cache of the application and redirect to login.

Reply:
Makes sense, thanks!